registry/config.go

```diff
@@ -89,45 +89,68 @@ func newServiceConfig(options ServiceOptions) *serviceConfig {
 		ServiceConfig: registrytypes.ServiceConfig{
 			InsecureRegistryCIDRs: make([]*registrytypes.NetIPNet, 0),
 			IndexConfigs:          make(map[string]*registrytypes.IndexInfo, 0),
-			// Hack: Bypass setting the mirrors to IndexConfigs since they are going away
-			// and Mirrors are only for the official registry anyways.
 		},
 		V2Only: options.V2Only,
 	}
 
-	config.LoadMirrors(options.Mirrors)
-	config.LoadInsecureRegistries(options.InsecureRegistries)
+	// Copy mirror information to IndexConfigs
+	err := config.LoadMirrors(options.Mirrors)
+	if err != nil {
+		panic(fmt.Sprintf("error loading mirrors: %s\n", err.Error()))
```

Comment:
FWIW, I don't think we panic in other cases like this; I don't think a panic is really suitable here; we validated the format, and the format was invalid, so the error should be printed, but a full dump